ci/generate_azure_pipelines_matrices.py

```diff
@@ -92,11 +92,13 @@ def __init__(self, platform, build_jobs):
 qt_versions = ["5.12.12", "5.15.14", "6.5.3"]
 
 linux_build_jobs = []
+linux_arm64_build_jobs = []
 mac_build_jobs = []
 windows_build_jobs = []
 
 all_platform_build_jobs = [
     PlatformBuildJobs("linux", linux_build_jobs),
+    PlatformBuildJobs("linux_arm64", linux_arm64_build_jobs),
```

Comment:
I don't think it is valuable  test it for all versions of targets.

Reply:
I'm not clear how you want to limit it - should `PlatformBuildJobs` take a subset of python versions or something? It looks like `PlatformBuildJobs` is how we get variables into matrices which is used in `azure-pipelines.yml` to actually run tests via `steps.yml`? I'm not really familiar with azure pipelines so I was just following how the other platforms work, and linux_arm64 is a new platform.

Reply:
Sorry for my misunderstanding.  My comment here is incorrect.

> I'm not clear how you want to limit it - should `PlatformBuildJobs` take a subset of python versions or something? It looks like `PlatformBuildJobs` is how we get variables into matrices which is used in `azure-pipelines.yml` to actually run tests via `steps.yml`? I'm not really familiar with azure pipelines so I was just following how the other platforms work, and linux_arm64 is a new platform.

A code chunk
```python
print("Setting Variables below")
print(
    f"##vso[task.setVariable variable=linux;isOutput=true]{json.dumps(matrices['linux'])}"
)
```
defines output matrix variable "linux"  that passed to `azure-pipelines.yml` 
then Linux tasks are defined as
```yaml
- job: Linux
  dependsOn: MatricesGenerator
  pool:
    vmImage: 'ubuntu-22.04'
  strategy:
    matrix: $[ dependencies.MatricesGenerator.outputs['mtrx.linux'] ]
  steps:
    - template: ci/steps.yml
```

Here it refers MatirxGenerator output `linux` variable.
The actual steps are defined in `ci/steps.yml`

 `PlatformBuildJobs("linux_arm64"` defines a definitions of platform "linux_arm64" and output as json.
A defined jobs will be referenced like

```yaml
  strategy:
    matrix: $[ dependencies.MatricesGenerator.outputs['mtrx.linux_arm64'] ]
```
 in `azure-pipelines.yml`.

Reply:
> Sorry for my misunderstanding. My comment here is incorrect.

I'm not clear if you are still asking for changes or not.

> A code chunk
> 
> ```python
> print("Setting Variables below")
> print(
>     f"##vso[task.setVariable variable=linux;isOutput=true]{json.dumps(matrices['linux'])}"
> )
> ```
> 
> defines output matrix variable "linux" that passed to `azure-pipelines.yml` 

I am doing this for linux_arm64 here https://github.com/rectalogic/aqtinstall/blob/b51d200c32fd948f6e918c5ecada1dc621e68161/ci/generate_azure_pipelines_matrices.py#L486-L488

> then Linux tasks are defined as
> 
> ```yaml
> - job: Linux
>   dependsOn: MatricesGenerator
>   pool:
>     vmImage: 'ubuntu-22.04'
>   strategy:
>     matrix: $[ dependencies.MatricesGenerator.outputs['mtrx.linux'] ]
>   steps:
>     - template: ci/steps.yml
> ```

I am doing this for linux_arm64 here https://github.com/rectalogic/aqtinstall/blob/b51d200c32fd948f6e918c5ecada1dc621e68161/azure-pipelines.yml#L57-L67
This is where I use a docker container on azure to run arm64 platform (not sure this will work, needs testing)


> `PlatformBuildJobs("linux_arm64"` defines a definitions of platform "linux_arm64" and output as json. A defined jobs will be referenced like
> 
> ```yaml
>   strategy:
>     matrix: $[ dependencies.MatricesGenerator.outputs['mtrx.linux_arm64'] ]
> ```
> 
> in `azure-pipelines.yml`.

https://github.com/rectalogic/aqtinstall/blob/b51d200c32fd948f6e918c5ecada1dc621e68161/azure-pipelines.yml#L65

Reply:
I see. Now I can understand and LGTM.